### PR TITLE
chore: added transport to RawAxiosRequestConfig

### PR DESCRIPTION
**Pull request by Cadienvan**
Related, yet separated from #5444 , added transport property to the RawAxiosRequestConfig. As it isn't checked in input, can't really provide anything but `any`.

### CODE REVIEW

**Comment by Cadienvan** [comment]
I'm pretty sure there was a flaky test, it isn't related to my commit.

Can someone re-run tests?

**Comment by yanfaliupst** [comment]
bump? I'm also looking for this field and I would like it to be in the d.ts file as it's documented on the README.md. Thanks

File: index.d.ts [hunk 1]
```diff
@@ -331,6 +331,7 @@ export interface AxiosRequestConfig<D = any> {
   maxRate?: number | [MaxUploadRate, MaxDownloadRate];
   beforeRedirect?: (options: Record<string, any>, responseDetails: {headers: Record<string, string>}) => void;
   socketPath?: string | null;
+  transport?: any;
```
Comment (DigitalBrainJS) [inline index.d.ts:334]: This line must also be added to [index.d.сts](https://github.com/axios/axios/blob/v1.x/index.d.cts) in order to have consistent types for the CommonJS module.

**Review by DigitalBrainJS — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (yanfaliupst) [reply index.d.ts:334]: @Cadienvan bump?

Comment (Cadienvan) [reply index.d.ts:334]: Yep I'll do it tomorrow sorry guys it's Easter days 🤣

**Comment by Cadienvan** [comment]
There you go @DigitalBrainJS @yanfaliupst   :)

**Review by DigitalBrainJS — APPROVED** [review]

**Comment by yanfaliupst** [comment]
Thanks so much!

**Pull request merged**